Source/WebKit/UIProcess/WebFullScreenManagerProxy.cpp

```diff
@@ -316,10 +300,26 @@ void WebFullScreenManagerProxy::beganEnterFullScreen(const IntRect& initialFrame
     });
 }
 
-void WebFullScreenManagerProxy::beganExitFullScreen(const IntRect& initialFrame, const IntRect& finalFrame)
+void WebFullScreenManagerProxy::beganExitFullScreen(const IntRect& initialFrame, const IntRect& finalFrame, CompletionHandler<void()>&& completionHandler)
 {
-    if (CheckedPtr client = m_client)
-        client->beganExitFullScreen(initialFrame, finalFrame);
+    CheckedPtr client = m_client;
+    if (!client)
+        return completionHandler();
+    client->beganExitFullScreen(initialFrame, finalFrame, [this, protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler)] mutable {
+        m_fullscreenState = FullscreenState::NotInFullscreen;
+        RefPtr page = m_page.get();
+        if (!page)
+            return completionHandler();
+        ALWAYS_LOG(LOGIDENTIFIER);
```

Comment:
Nit: it is better to pass a variable initialized with `LOGIDENTIFIER` into the completion handler because `__func__` inside of a completion handler doesn't generate a method name that makes sense for our logging.